beacon_node/network/src/router/processor.rs

```diff
@@ -183,61 +183,30 @@ impl<T: BeaconChainTypes> Processor<T> {
         }
     }
 
-    /// Process a `Status` message, requesting new blocks if appropriate.
-    ///
-    /// Disconnects the peer if required.
+    /// Process a `Status` message to determine if a peer is relevant to us. Irrelevant peers are
+    /// disconnected; relevant peers are sent to the SyncManager
     fn process_status(
         &mut self,
         peer_id: PeerId,
-        status: StatusMessage,
+        remote: StatusMessage,
     ) -> Result<(), BeaconChainError> {
-        let remote = PeerSyncInfo::from(status);
-        let local = match PeerSyncInfo::from_chain(&self.chain) {
-            Some(local) => local,
-            None => {
-                error!(
-                    self.log,
-                    "Failed to get peer sync info";
-                    "msg" => "likely due to head lock contention"
-                );
-                return Err(BeaconChainError::CannotAttestToFutureState);
-            }
-        };
-
+        let local = status_message(&self.chain)?;
         let start_slot = |epoch: Epoch| epoch.start_slot(T::EthSpec::slots_per_epoch());
 
-        if local.fork_digest != remote.fork_digest {
-            // The node is on a different network/fork, disconnect them.
-            debug!(
-                self.log, "Handshake Failure";
-                "peer_id" => peer_id.to_string(),
-                "reason" => "incompatible forks",
-                "our_fork" => hex::encode(local.fork_digest),
```

Comment:
It has been quite often that this error happens with clients on different testnets/configurations or buggy clients. 

Printing the two forks has historically been very valuable. And it would be good to include this in the error, I think

Reply:
I updated it :+1: